File: mobile/lib/domain/entities/content_suggestion.dart
enum SuggestionStatus { pending, accepted, rejected, edited }

class ContentSuggestion {
  final String id;
  final String suggestedText;
  final String rationale;
  final String riskAssessment;
  final String estimatedImpact;
  final DateTime generatedAtUtc;
  SuggestionStatus status;

  ContentSuggestion({
    required this.id,
    required this.suggestedText,
    required this.rationale,
    required this.riskAssessment,
    required this.estimatedImpact,
    required this.generatedAtUtc,
    this.status = SuggestionStatus.pending,
  });

  static SuggestionStatus _parseStatus(dynamic v) {
    if (v is String) {
      switch (v.toLowerCase()) {
        case 'accepted':
          return SuggestionStatus.accepted;
        case 'rejected':
          return SuggestionStatus.rejected;
        case 'edited':
          return SuggestionStatus.edited;
        default:
          return SuggestionStatus.pending;
      }
    }
    if (v is int) {
      if (v == 1) return SuggestionStatus.accepted;
      if (v == 2) return SuggestionStatus.rejected;
      if (v == 3) return SuggestionStatus.edited;
      return SuggestionStatus.pending;
    }
    return SuggestionStatus.pending;
  }

  factory ContentSuggestion.fromJson(Map<String, dynamic> json) {
    return ContentSuggestion(
      id: json['id'],
      suggestedText: json['suggestedText'] ?? '',
      rationale: json['rationale'] ?? '',
      riskAssessment: json['riskAssessment'] ?? 'Low',
      estimatedImpact: json['estimatedImpact'] ?? 'Unknown',
      generatedAtUtc:
          DateTime.tryParse((json['generatedAtUtc'] ?? '').toString()) ??
          DateTime.now().toUtc(),
      status: _parseStatus(json['status']),
    );
  }
}

File: src/TX_Manager.Domain/Entities/ContentSuggestion.cs
using System;
using TX_Manager.Domain.Enums;

namespace TX_Manager.Domain.Entities;

public class ContentSuggestion : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string SuggestedText { get; set; } = string.Empty; 
    public string Rationale { get; set; } = string.Empty; 
    
    public string RiskAssessment { get; set; } = "Low"; // Low, Medium, High
    public string EstimatedImpact { get; set; } = "Unknown"; // e.g. "High Engagement potential"

    public SuggestionStatus Status { get; set; } = SuggestionStatus.Pending;
    public string? RejectionReason { get; set; } 

    public DateTime GeneratedAt { get; set; } = DateTime.UtcNow;

    public Guid? ScheduledPostId { get; set; }
    public Post? ScheduledPost { get; set; }
}
